src/jank/clojure/core.jank

```diff
@@ -431,6 +431,108 @@
                  #{ o }#
                );"))
 
+(def conj!)
+(def conj!
+  (fn* conj!
+    ([]
+     (transient []))
+    ([coll]
+     coll)
+    ([coll x]
+     (native/raw "__value = visit_object
+                  (
+                    [&](auto const typed_t, auto const head) -> object_ptr
+                    {
+                      using T = typename decltype(typed_t)::value_type;
+
+                        if constexpr(behavior::persistentable<T>)
+                        {
+                          return typed_t->cons_in_place(head);
+                        }
```

Comment:
Single statement expressions which can fit on one line get wrapped up like `{ this; }`. Check out the `else` right below.

Also, the `if` here is indented futher than the `using`. Both should be 2 spaces into the `{`.

Formatting of inline C++ is currently unsolved. I'd love a clang-format extension which can work with this and keep uniformity with our C++ code base. Until then, best to be consistent with what I have.